### PR TITLE
[FLOC-1088] Use mulitple github statuses.

### PR DESCRIPTION
**Pull request by tomprince**
https://clusterhq.atlassian.net/browse/FLOC-1088

The results can be seen [here](https://github.com/ClusterHQ/flocker/compare/test-commit).

### CODE REVIEW

File: flocker_bb/github.py [hunk 1]
```diff
@@ -35,9 +40,67 @@ def __init__(self, token):
...
+                        if isinstance(v, unicode)})
+        log.msg(format="github request %(request)s", request=request)
+        d = self._github.repos.createStatus(**request)
+        d.addErrback(
+            log.err,
+            'While sending start status to GitHub: ' + repr(request))
+
+    @staticmethod
+    def _simplifyBuilderName(name):
+        name = name.rpartition('flocker-')[2]
```
Comment (adamtheturtle) [inline flocker_bb/github.py]: Explain here why some builders don't start with `flocker-`.

**Comment by adamtheturtle** [comment]
The sample results look good, but is it suitably clear what is going on while some of the builds are in progress, and others perhaps are finished?

**Comment by adamtheturtle** [comment]
Perhaps separately, please document how to test changes to `github.py` without potentially causing problems with the Flocker GitHub project.

**Comment by adamtheturtle** [comment]
As a general point, the lack of maintainability helpers like those required in Flocker, particularly docstrings, make this slow to review and I guess harder to maintain. I think that now particularly now that  we have issues in the Flocker JIRA, and while this codebase is relatively small, we should consider adding those niceties like docstrings. If you think that is sensible, please file a follow up issue. Else, at least, please add a few docstrings in this code.

**Comment by adamtheturtle** [comment]
Please create any issues and make any changes as you see fit based on the above comments. Nothing here stands out as bad, but it is, as mentioned above, quite difficult to follow, for me at least. Merge, or resubmit if there are any major changes, thanks.

**Comment by tomprince** [comment]
> The sample results look good, but is it suitably clear what is going on while some of the builds are in progress, and others perhaps are finished?

I believe so.  When a build is submitted (either from a commit, or a force build form), it will show up as pending. Once the build has started, the status will be updated, along with a link to the build. Finally when the build is finished, the status is updated again.

One slight wrinkle is that some builds trigger others, but I think the new build should show up as pending before the old build finishes. Even if that isn't the case, there should be a small enough window between the two that it doesn't matter.

**Pull request merged**